Integrations/CentralBankIntegrationLib/CentralBankCurrenciesPublisher.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Interfaces;
+using System.Collections.Generic;
+using System.Net;
+using System.IO;
+using System.Xml.Serialization;
+using System.Text;
+using CentralBankIntegrationLib.XmlSerializationObjects;
+using System.Linq;
+using System;
+using DTO;
+using System.Threading.Tasks;
+
+namespace CentralBankIntegration
+{
+    /// <summary>
+    /// Get reference information from Central Bank.
+    /// </summary>
+    public class CentralBankCurrenciesPublisher : INewsPublisher
+    {
+        const string uri = "http://www.cbr.ru/scripts/XML_daily.asp";
+        DataFromCentralBank data;
+
+        public CentralBankCurrenciesPublisher(Uri uriBank)
+        {
+            data = GetData(uriBank);
+        }
+
+        public CentralBankCurrenciesPublisher() : this(new Uri(uri)) { }
+
+        public List<ExchangeRate> GetCurrencies()
+        {
+            List<ExchangeRate> rates = new List<ExchangeRate>();
+            //Parallel.For(0, data.Currencies.Count, (int i) =>
+            //{
+            //    rates.Add(new ExchangeRate { CurrencyCode = data[i].CharId, Value = data[i].DigitValue });
+            //});
+
+            for(int i = 0; i < data.Currencies.Count; i++)
+            {
+                rates.Add(new ExchangeRate { CurrencyCode = data[i].CharId, Value = data[i].DigitValue });
+            }
+
+            return rates;
+        }
+
+        // Connect to Central Bank Exchange Rates, get the most popular valutes.
+        private DataFromCentralBank GetData(Uri uri)
+        {
+            HttpWebRequest httpWebRequest = (HttpWebRequest)WebRequest.Create(uri);
+            HttpWebResponse httpWebResponse = (HttpWebResponse)httpWebRequest.GetResponse();
+            DataFromCentralBank data;
+            XmlSerializer serializer = new XmlSerializer(typeof(DataFromCentralBank));
+
+            using (Stream responseStream = httpWebResponse.GetResponseStream())
+            {
+                StreamReader streamReader = new StreamReader(responseStream);
```

Comment:
Not fit with code convention. How about var?

Reply:
I do not agree with this. There is nothing wrong with writing an object class

Reply:
there is nothing wrong with writing an object class. I do not agree

Reply:
But type of object is obvious. Why do we need this? And all of us agreed to write var in these cases

Reply:
not obvious to me

Reply:
Really? Object that created as  new StreamReader could be something else?